src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilterTest.java

```diff
@@ -461,6 +473,54 @@ public void testSuppressById() throws Exception {
             "11:18: "
                 + getCheckMessage(AbstractNameCheck.class,
                     MSG_INVALID_PATTERN, "ID", "^[a-z][a-zA-Z0-9]*$"),
+            "15:17: "
+                + getCheckMessage(AbstractNameCheck.class,
+                    MSG_INVALID_PATTERN, "DEF", "^[a-z][a-zA-Z0-9]*$"),
+            "16:17: "
+                + getCheckMessage(AbstractNameCheck.class,
+                    MSG_INVALID_PATTERN, "XYZ", "^[a-z][a-zA-Z0-9]*$"),
+        };
+
+        verifySuppressed(filterConfig,
+            getPath("InputSuppressWithNearbyCommentFilterById.java"),
+            expectedViolationMessages, suppressedViolationMessages);
+    }
+
+    @Test
+    public void tesSuppressByIdAndMessage() throws Exception {
```

Comment:
@rnveach here is a new test case for the issue you identified. This test considers two possible violations:
```java
private int DEF = 4; // @cs-: ignore (allow DEF)
private int XYZ = 3; // @cs-: ignore (allow xyz)
```
they both match the ID (`"ignore"`) but only the first matches the message (`"Name 'DEF' must match pattern..."`), so the XYZ violation isn't suppressed.